lang/backend/src/ir/decls.rs

```diff
@@ -0,0 +1,110 @@
+//! High-level intermediate respresentation of the AST after erasure.
+//! This representation is shared between any compiler backends and hence can only make few assumptions about the compilation target.
+
+use codespan::Span;
+use url::Url;
+
+use ast::{Attributes, IdBind, UseDecl, VarBind};
+
+use super::exprs::{Case, Exp};
+use super::kind::Kind;
+
+#[derive(Debug, Clone)]
+pub struct Module {
+    pub uri: Url,
+    pub use_decls: Vec<UseDecl>,
+    pub decls: Vec<Decl>,
+}
+
+#[derive(Debug, Clone)]
+pub enum Decl {
```

Comment:
I would split Data/Codata from Def/Codef/Let. We usually don't generate code for the former, only for the latter, so they should be separated. We only consult Data/Codata declarations when it comes to generating code.

Reply:
Or just remove the `Decl` type and have separate struct fields in the `Module` struct.